central/jwt/jwt.go

```diff
@@ -52,6 +56,35 @@ func GetPrivateKeyBytes() ([]byte, error) {
 	return getBytesFromPem(privateKeyPathPEM)
 }
 
+type m2mValidator struct {
+	m2m.TokenExchangerSet
+	roxValidator tokens.Validator
+	issuerID     string
+}
+
+// Validate implements tokens.Validator.
+func (v *m2mValidator) Validate(ctx context.Context, token string) (*tokens.TokenInfo, error) {
+	if !v.HasExchangersConfigured() {
+		return v.roxValidator.Validate(ctx, token)
+	}
+	iss, err := m2m.IssuerFromRawIDToken(token)
+	if err != nil {
+		return nil, err
+	}
+	if iss == v.issuerID {
```

Comment:
Let's add a few comments on when and why we short-circuit here, so the token exchange part is more in your face.